sympy/core/relational.py

```diff
@@ -1138,3 +1060,169 @@ def _eval_relation(cls, lhs, rhs):
     '<': StrictLessThan,
     'lt': StrictLessThan,
 }
+
+
+def _n2(a, b):
+    """Return (a - b).evalf(2) if a and b are comparable, else None.
+    This should only be used when a and b are already sympified.
+    """
+    # /!\ it is very important (see issue 8245) not to
+    # use a re-evaluated number in the calculation of dif
+    if a.is_comparable and b.is_comparable:
+        dif = (a - b).evalf(2)
+        if dif.is_comparable:
+            return dif
+
+
+@dispatch(Basic, Basic)
+def _eval_is_le(lhs, rhs):
+    return None
+
+
+@dispatch(Basic, Basic)
+def _eval_is_ge(lhs, rhs):
+    return None
+
+
+@dispatch(Basic, Basic)
+def _eval_Eq(lhs, rhs):
+    return None
+
+
+
+def is_lt(lhs, rhs):
+    return fuzzy_not(is_ge(lhs, rhs))
+
+
+def is_gt(lhs, rhs):
+    return fuzzy_not(is_le(lhs, rhs))
+
+
+def is_le(lhs, rhs):
+    return is_ge(rhs, lhs)
+
+
+def is_ge(lhs, rhs):
+    retval = _eval_is_ge(lhs, rhs)
+    if retval is not None:
+        return retval
+    else:
+        n2 = _n2(lhs, rhs)
+        if n2 is not None:
+            # use float comparison for infinity.
+            # otherwise get stuck in infinite recursion
+            if n2 in (S.Infinity, S.NegativeInfinity):
+                n2 = float(n2)
+            return _sympify(n2 >= 0)
+        if lhs.is_extended_real and rhs.is_extended_real:
+            if (lhs.is_infinite and lhs.is_extended_positive) \
+                     or (rhs.is_infinite and rhs.is_extended_negative):
+                return S.true
+            diff = lhs - rhs
+            if diff is not S.NaN:
+                rv = diff.is_extended_nonnegative
+                if rv is not None:
+                   return rv
+
+
+def is_neq(lhs, rhs):
+    from .logic import fuzzy_not
+    return fuzzy_not(is_eq(lhs, rhs))
+
+
+def is_eq(lhs, rhs):
+    from sympy.core.add import Add
+    from sympy.functions.elementary.complexes import arg
+    from sympy.simplify.simplify import clear_coefficients
+    from sympy.utilities.iterables import sift
+
+    retval = _eval_Eq(lhs, rhs)
+    if retval is not None:
+        return retval
```

Comment:
I think this method will still need to check e.g. `lhs._eval_Eq` for backward compatibility. Users and downstream libraries have probably defined `_eval_Eq` methods and we need to try to ensure that their code would still work with these changes.